### ISSUE TITLE
Indirect Band Gap Functionality

### ISSUE THREAD

**Issue by kavanase** [WMD-group/TrapLimitedConversion#2]
Just something to keep in mind for development.

I think it would be good to allow specification of an indirect band gap, as currently only a single (assumed direct) band gap can be provided. I think it should only require a minor update to the code, by using the indirect gap value for the cooled carrier energies. This would extend the functionality of `aTLC` to indirect gap materials, which would be useful both when assuming step-like absorption (i.e. SQ limit efficiencies) and finite absorption with aTLC.

Also just to note on this point, I think it would be important to be clear in the docstrings/tutorials how the input absorption and band gap values are used (i.e. if the onset of absorption differs from `E_gap`, is it shifted to match or not? How else is `E_gap` used etc).

**Comment by xw-w** [comment]
For systems with indirect band gaps, I guess it's better to use indirect band gaps as inputs

But thanks for the suggestions and I'll improve the relevant tutorials

**Comment by kavanase** [comment]
Yes I agree, but it seems in the current code this makes no difference when using TLC/aTLC (with `l_sq = False`):
<img width="696" alt="image" src="https://github.com/user-attachments/assets/411554de-f429-4772-9658-cb83945e6fb2">
(Which can also be seen from the code)

It assumes the band gap is direct, and determines it from the input absorption file, so setting `E_gap` to less than the direct/absorption band gap makes no difference to the efficiency, when it should significantly decrease it. In theory, it should be relatively easy to extend the code to handle these cases, so just noting in case there is some future development of the package!

Setting `E_gap` to higher than the absorption band gap does decrease the efficiency as expected, I think because it cuts off the absorption below this point:
<img width="623" alt="image" src="https://github.com/user-attachments/assets/4d3a92c0-e542-4e8f-a754-235a7c881796">
